compiler-core/src/language_server/engine.rs

```diff
@@ -908,6 +909,62 @@ fn position_within(position: lsp_types::Position, range: lsp_types::Range) -> bo
     position >= range.start && position <= range.end
 }
 
+fn code_action_unused_values(
+    module: &Module,
+    params: &lsp::CodeActionParams,
+    actions: &mut Vec<CodeAction>,
+) {
+    let uri = &params.text_document.uri;
+    let unused: Vec<&SrcSpan> = module
+        .ast
+        .type_info
+        .warnings
+        .iter()
+        .filter_map(|warning| match warning {
+            type_::Warning::ImplicitlyDiscardedResult { location } => Some(location),
+            _ => None,
+        })
+        .collect();
+
+    if unused.is_empty() {
+        return;
+    }
+
+    // Convert src spans to lsp range
+    let line_numbers = LineNumbers::new(&module.code);
+
+    for unused in unused {
+        let SrcSpan { start, end } = *unused;
+
+        let hover_range = src_span_to_lsp_range(SrcSpan::new(start, end), &line_numbers);
+
+        let start_char = module
+            .code
+            .get(start as usize..start as usize + 1)
+            .unwrap_or_default();
+
+        // only show actions for the most inner span
+        if (start_char == "{"
+            && params.range.start.line != hover_range.start.line
+            && params.range.end.line != hover_range.end.line)
+            || !overlaps(params.range, hover_range)
+        {
+            continue;
+        }
```

Comment:
What does this do? It says "only show actions for the most inner span", but it seems to be checking if the first character is a `{`, not if it is the inner-most span.

Reply:
fixed now!